paymentsheet/src/test/java/com/stripe/android/paymentsheet/SavedPaymentMethodMutatorTest.kt

```diff
@@ -23,6 +24,140 @@ import org.mockito.Mockito.mock
 import org.mockito.kotlin.whenever
 
 class SavedPaymentMethodMutatorTest {
+    @Test
+    fun `canRemove is correct when no payment methods for customer`() = runScenario(
+        allowsRemovalOfLastSavedPaymentMethod = true,
+    ) {
+        savedPaymentMethodMutator.canRemove.test {
+            assertThat(awaitItem()).isFalse()
+
+            customerStateHolder.setCustomerState(
+                CustomerState.createForLegacyEphemeralKey(
+                    customerId = "cus_123",
+                    accessType = PaymentSheet.CustomerAccessType.LegacyCustomerEphemeralKey("ek_123"),
+                    paymentMethods = listOf()
+                )
+            )
+
+            ensureAllEventsConsumed()
```

Comment:
I thought using `test` made this assertion internally.

Reply:
I think it does but I added it to make the test more clear on what it is doing at the end rather than do no assertions since `StateFlow` does not repost the same value.

Reply:
Ahh, I see. Might be worth a comment then (the value didn't change). As is, I don't think it provides a ton of clarity. But I agree it would with the assertion + a comment.